fw/sock.c

```diff
@@ -654,8 +657,15 @@ ss_do_close(struct sock *sk)
 	}
 	if (sk->sk_state == TCP_CLOSE) {
 		struct request_sock *req = tcp_sk(sk)->fastopen_rsk;
-		if (req != NULL)
+		if (req)
```

Comment:
As i understand it, this function kind of resembles the `tcp_close()` from `net/ipv4/tcp.c`.
The code here is quite old and i'm wondering if we should recheck and update it probably?
For example, `TFO` stuff is RCU protected now, as can be seen in `tcp_close()` code.

Reply:
Yeah, I'll review the code. BTW it seems the mainline still has the integer variable `data_was_unread` used as `bool` only https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/net/ipv4/tcp.c#n2855 . I appreciate it you prepare and submit the patch to LKML.